utils/HACKING.adoc

```diff
@@ -0,0 +1,52 @@
+== Magic numbers
+
+The magic numbers in `config.mlp` are included in the header of
+compiled files produced by the OCaml compiler. Different kind of files
+(cmi, cmo, cmx, cma, executables, etc.) get different magic numbers,
+and we also change the magic number whenever we change the format of
+the corresponding file.
+
+This lets the compiler differenciates files that should be valid files
+of the kind it expects, and files that are passed by mistake, either
+that are not at all valid compiled files, or because they come from
+a different compiler version with an incompatible file format.
+
+We say that we "bump" a magic number when we update its version part
+in config.mlp. To bump all magic numbers is to increment the version
+of all kinds of magic numbers.
```

Comment:
You should write "every kind", as "all kinds of" has another meaning in English.